4.1.15 release branch

dbeltrankyl opened the pull request:
Autosubmit 4.1.15 release. 

As mentioned in slack, cutting developments to include some features like job_metrics and stats fixes. 

I've cherry-picked two commits, of which I already know that one is not working fine.

**Check List**

- [x] Test the setstatus fixes ( no pytest for this one, as it is being changed for #2211 
- [x] Test the tmpdir ps platform with a shared filesystem fix
  - [x] Works in /esarchive ps platforms
  - [x] Works in remote PS platforms
  - [ ] Add pytest
- [ ] Tsuite OK ( module 4.1.15-dev-#last_commit )
- [ ] Deployment in:
  - [ ] Git
  - [ ] Pip
  - [ ] Newhubs
  - [ ] climate-dt
  - [ ] container
 - [ ] Bump issues

dbeltrankyl commented:
After rebasing, I tried again. Same issue, in the pipeline works and in my machine it works 1/3 times

kinow commented:
```bash
(autosubmit4) kinow@ranma:~/Development/python/workspace/autosubmit$ git log -n 1
commit 76de04f88fa7f4cf0a0c7872be8d1d8cab1c6559 (HEAD -> 4.1.15-release-branch, upstream/4.1.15-release-branch)
Author: dbeltrankyl <daniel.beltran@bsc.es>
Date:   Fri Jun 13 13:05:35 2025 +0200

    updated test
```

I executed the `sudo` command in the `CONTRIBUTING.md` to fix the Docker commands (I always need to run those to use testcontainers), then launched the unit tests with `pytest -m ''`.

```bash
========================================================================================================== 617 passed, 26 skipped, 2 xfailed, 4 xpassed, 100 warnings in 7.37s ===========================================================================================================
```

Then did the same again for integration tests, `pytest -m '' test/integration`.

```bash
===================================================================================================================== 94 passed, 20 skipped, 441 warnings in 54.46s ======================================================================================================================
```

Then regression tests,

```bash
============================================================================================================================ 33 passed, 42 warnings in 2.20s =============================================================================================================================
```

I think I am not able to reproduce the issue you are having. If you have a specific test, or command line that you are testing, I can try these on my laptop if you'd like.

kinow commented:
Great! Let me know if you need any help testing the release :+1: Thanks Dani!

kinow commented:
Just merged that PR for the error log level (thanks Erick). Nothing more from me that could go in 4.1.15. From tomorrow onward I'll be on 4.1.16 :grimacing:

dbeltrankyl commented:
And tomorrow morning (8:00am~~) I'll create the module

dbeltrankyl commented:
```
[dbeltran@climatedt-wf ~]$ autosubmit expid -dm -H "local" -d "test"
/appl/AS/4.1.15-dev-55d4fe3/lib64/python3.9/site-packages/networkx/utils/backends.py:135: RuntimeWarning: networkx backend defined more than once: nx-loopback
  backends.update(_get_backends("networkx.backends"))
Autosubmit is running with 4.1.15
```

Seems that is only a warning, but I didn't see it before cc @kinow 

I'll go to the tsuite issue and see if they can start to use this one

dbeltrankyl commented:
Hello @kinow, 

I did a quick fix for https://earth.bsc.es/gitlab/digital-twins/de_340-2/workflow/-/issues/1118#note_376828
```
    def _substitute_placeholders(
            self,
            content: str,
            parameters: dict,
            as_conf: AutosubmitConfig,
            undefined_variables: list[str] = None
    ) -> str:
        ...
        # TODO quick fix for 4.1.15 release, to see why it is needed
        if as_conf.dynamic_variables:
            parameters = as_conf.substitute_dynamic_variables(parameters, max_deep=25, in_the_end=True)
        ...
```
 
I think it's due to the way we're calling update_parameters, but I'm not sure why. 

The reason for the quick fix is:

- Solves our current issue.
- Not enough time to do a proper fix.
- In 4.2.0 there are already changes due the job_list.db so I think it is a waste of time to do changes in 4.1.X related to update_parameters

dbeltrankyl commented:
Creating a new module for the tsuite, I hope this is the last one

kinow commented:
Hi @dbeltrankyl , I saw the last commit and it was really simple. Thanks for adding that to the changelog. Over next days I will keep going through bugs, update anything that was fixed, and maybe try to write pending tests for some changes (I think we have a few TODO's in some places, I will check if @VindeeR can help with that too over summer break).

Hopefully the last module :crossed_fingers: , otherwise, I can take over the troubleshooting and support, update & test this branch, and release a new module to ClimateDT for the testing suite :+1: so do not push yourself too hard as I think you are about to start your holidays soon. You showed me how to create `module`, how to troubleshoot most issues, and I'm following this branch as close as I can :+1: 

Thanks, Dani!

dbeltrankyl commented:
Thanks!

dbeltrankyl commented:
v4.1.15 was created https://github.com/BSC-ES/autosubmit/releases/tag/4.1.15

After that I rebased and added a commit to fix a test incompatibility. 

waiting for the pipeline before merging
